src/benchmarks/clsparse-bench/functions/clfunc_xSpMSpM.hpp

```diff
@@ -0,0 +1,283 @@
+#define CLSPARSE_BENCHMARK_SpM_SpM_HXX
+
+#include "clSPARSE.h"
+#include "clfunc_common.hpp"
+
+// Dummy Code to be deleted later
+clsparseStatus clsparseDcsrSpGemm(const clsparseCsrMatrix* sparseMatA, const clsparseCsrMatrix* sparseMatB, clsparseCsrMatrix* sparseMatC, const clsparseControl control)
+{
+    std::cout << "sparseMat A dimensions = \n";
+    std::cout << " rows = " << sparseMatA->num_rows << std::endl;
+    std::cout << " cols = " << sparseMatA->num_cols << std::endl;
+    std::cout << "nnz = " << sparseMatA->num_nonzeros << std::endl;
+
+    std::cout << "sparseMat B dimensions = \n";
+    std::cout << " rows = " << sparseMatB->num_rows << std::endl;
+    std::cout << " cols = " << sparseMatB->num_cols << std::endl;
+    std::cout << "nnz = " << sparseMatB->num_nonzeros << std::endl;
+
+    return clsparseSuccess;
+}//
+// Dummy code ends
+
+template <typename T>
+class xSpMSpM : public clsparseFunc {
+public:
+    xSpMSpM(PFCLSPARSETIMER sparseGetTimer, size_t profileCount, cl_device_type devType) : clsparseFunc(devType, CL_QUEUE_PROFILING_ENABLE), gpuTimer(nullptr), cpuTimer(nullptr)
+    {
+        //	Create and initialize our timer class, if the external timer shared library loaded
+        if (sparseGetTimer)
+        {
+            gpuTimer = sparseGetTimer(CLSPARSE_GPU);
+            gpuTimer->Reserve(1, profileCount);
+            gpuTimer->setNormalize(true);
+
+            cpuTimer = sparseGetTimer(CLSPARSE_CPU);
+            cpuTimer->Reserve(1, profileCount);
+            cpuTimer->setNormalize(true);
+
+            gpuTimerID = gpuTimer->getUniqueID("GPU xSpMSpM", 0);
+            cpuTimerID = cpuTimer->getUniqueID("CPU xSpMSpM", 0);
+        }
+        clsparseEnableAsync(control, false);
+    }
+
+    ~xSpMSpM() {}
+
+    void call_func()
+    {
+        if (gpuTimer && cpuTimer)
+        {
+            gpuTimer->Start(gpuTimerID);
+            cpuTimer->Start(cpuTimerID);
+
+            xSpMSpM_Function(false);
+
+            cpuTimer->Stop(cpuTimerID);
+            gpuTimer->Stop(gpuTimerID);
+        }
+        else
+        {
+            xSpMSpM_Function(false);
+        }
+    }//
+
+    double gflops()
+    {
+        return 0.0;
+    }
+
+    std::string gflops_formula()
+    {
+        return "N/A";
+    }
+
+    double bandwidth() // Need to modify this later **********
+    {
+        //  Assuming that accesses to the vector always hit in the cache after the first access
+        //  There are NNZ integers in the cols[ ] array
+        //  You access each integer value in row_delimiters[ ] once.
+        //  There are NNZ float_types in the vals[ ] array
+        //  You read num_cols floats from the vector, afterwards they cache perfectly.
+        //  Finally, you write num_rows floats out to DRAM at the end of the kernel.
+        return (sizeof(cl_int)*(csrMtx.num_nonzeros + csrMtx.num_rows) + sizeof(T) * (csrMtx.num_nonzeros + csrMtx.num_cols + csrMtx.num_rows)) / time_in_ns();
```

Comment:
Yes we have an idea it based on nnzCt_full = total number of non-zeros present in output temporary matrix, unfortunately this parameter is a local variable hidden inside the API. Hence we couldn't use this in the current state of the code. After I modify the code I will try to use this in Flops calculation.

But wondering what flops calculation I should be using for cusparse because they use different algorithm.